readthedocs/projects/tasks.py

```diff
@@ -1002,15 +1002,14 @@ def symlink_project(project_pk):
 
 
 @app.task(queue='web')
-def symlink_domain(project_pk, domain_pk, delete=False):
+def symlink_domain(project_pk, domain_str, delete=False):
```

Comment:
Same note able annotating variable name with type. `domain` is probably best, as `domain_pk` doesn't make sense here anymore. Is there a strong reason to change this call either way? Why not change the call to `sym.remove_symlink_cname(domain.domain)` instead? We have calls from our commercial hosting fork that might break because of the signature change.

Reply:
@agjohnson 
I think we can't call like this `sym.remove_symlink_cname(domain.domain)` because as mentioned in the issue ( #4789 ).

> This task tries to get the Domain object using the pk from the database and it fails because the object was already removed.

So it needs to depend only on the string `domain.domain`.